Chore: Fixing typos

brianmuse opened the pull request:
This is a followup to #8 which I prematurely, mistakenly merged to correct typos pointed out by @zachdunn.

I changed the `Note!`s too. For the abandoned ones the `@deprecated` tag should be enough. For the open SSL ones that are insecure i changed `Note!` to `Insecure!` to be a little more clear about why we're exclaiming `!`.

brianmuse committed:
Fixing typos

jstruzik commented:
LGTM 👍 

You're very good at words.

Rican7 commented on src/Robin/Ntlm/Crypt/Random/McryptRandomByteGenerator.php:
```diff
@@ -16,7 +16,7 @@
  *
  * @link http://php.net/mcrypt
- * @deprectated NOTE! This implementation is deprecated, as the mcrypt library
+ * @deprecated This implementation is deprecated, as the mcrypt library
```
Lol, I typed it correctly in one place but not another. Sick.

Rican7 commented on src/Robin/Ntlm/Crypt/Random/OpenSslRandomByteGenerator.php:
```diff
@@ -15,7 +15,7 @@
  *
  * @link http://php.net/openssl
- * @deprectated NOTE! This implementation is deprecated, as it's been found to
+ * @deprecated Insecure! This implementation is deprecated, as it's been found to
```
Why the change to "insecure!"? The " NOTE!" is a common "task tag" (along with "TODO" and "FIXME") that gets automatically highlighted by many text editors. Changing it to " insecure!" removes that functionality.

brianmuse replied:
It's not a tag, though. It's a message for the deprecation. If we wanted to move this to the docblock body, `NOTE:` would make more sense I think.

brianmuse replied:
Also is this an example task tag? The deprecation tag itself is that. This is more FYI on why it's being deprecated and why you shouldn't use it.

Rican7 replied:
Huh? The `@deprecated` bit is a doc tag. The "NOTE!" is a task tag, which can go anywhere in a comment body to denote importance via highlighting.

Rican7 replied:
I'd stick to simply fixing the spelling mistake, as I would have done if it hasn't had been prematurely merged. This other change is unnecessary.

brianmuse replied:
Can you explain why a NOTE or TODO is needed within a deprecation tag? I've never seen that. I was intending to make this comment on the original PR had there been other changes to make. 

The deprecation tag explains that it's deprecated and why its deprecated. If a further todo is needed, such as `@todo Remove due to insecurity`, that should go in the main docblock.

Otherwise you could make the case that every `@deprecation` tag ever warrants a `NOTE:` first. It's just a bit redundant.

brianmuse replied:
While I'm thinking about this, the deprecation tag description should probably offer some alternative, such as `... Use xyz instead.`

Rican7 replied:
> Otherwise you could make the case that every `@deprecation` tag ever warrants a `NOTE:` first. It's just a bit redundant. 

Good point. Yea. :/

> While I'm thinking about this, the deprecation tag description should probably offer some alternative, such as `... Use xyz instead.`

Another good point. I believe I linked to an issue, but you're right, usually you want to point to an implementation that should be moved to or used instead.

brianmuse replied:
Lemme tidy up with todo in the body and a suggested alternative in the deprecation. Best of both worlds.

brianmuse replied:
https://github.com/robinpowered/php-ntlm/pull/9/commits/e5e6625ab5cb7d2ca934a7032a56ec74a54331b4

Rican7 replied:
Great compromise. I'm into it. 👍

brianmuse committed:
Updating comments for deprecations.

The `@todo` tag is used for internal messages for developers maintaining the library and the class, explaining that (and why) it should be removed in a future version. The `@deprecated` tag is intended for the eyes of someone using this library to tell them that the method is deprecated and giving an alternative suggestion.

Rican7 commented:
This LGTM 👍

jstruzik commented:
👍
